bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiRpc.java

```diff
@@ -491,13 +491,24 @@ protected void installScript(String script, boolean install) throws ShellyApiExc
             if (upload) {
                 logger.debug("{}: Script will be installed...", thingName);
 
-                // Create new script, get id
-                ShellyScriptResponse rsp = apiRequest(
-                        new Shelly2RpcRequest().withMethod(SHELLYRPC_METHOD_SCRIPT_CREATE).withName(script),
-                        ShellyScriptResponse.class);
-                ourId = rsp.id;
-                logger.debug("{}: Script has been created, id={}", thingName, ourId);
-                upload = true;
+                if (ourId == -1) {
+                    // find free script id
+                    ourId = 0;
+                    while (++ourId < 10 && testScriptId(scriptList, ourId)) {
```

Comment:
Magic number and repeated, consider extracting to a constant:
```java
private static final int MAX_SCRIPTS = 10;
```

Consider rewriting the loop for clarity and add a comment in the empty body, e.g.:
```suggestion
                    for (ourId = 1; ourId < MAX_SCRIPTS && testScriptId(scriptList, ourId); ourId++) {
                        // find free script id
```

Reply:
changed

Reply:
And make the constant `final` as well?

Reply:
And rename the constant, unless it really means that ID 10 is allowed?

Reply:
yes, IDs start with 1 and will be incremented. Plus supports only 3, but Pro up to 10 scripts
I now changed MAX_SCRIPT_ID to 15